src/communication/mpi.hpp

```diff
@@ -106,9 +106,11 @@ namespace mpi {
         std::vector<char> buffer(displs.back());
 
         PE("MPI", "Gather");
-        MPI_Gatherv(str.data(), counts[rank()], traits::mpi_type(),                  // send
-                    buffer.data(), counts.data(), displs.data(), traits::mpi_type(), // receive
-                    root, MPI_COMM_WORLD);
+        MPI_Gatherv(
+            // const_cast required for MPI implementations that don't use const* in their interfaces
```

Comment:
Could also #define MPICH2_CONST (See CMake for for BGQ)

Reply:
This was to work around an ancient version of OpenMPI provided by Ubuntu/TravisCI.